src/ImageStream.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Diactoros;
+
+use GdImage;
+
+use function sprintf;
+
+/**
+ * This class purposely does not override the $resource property in order to
+ * allow it to extend Stream. If defined here with a type, PHP will raise a
+ * fatal error complaining that it must not define a type for the property.
+ */
+class ImageStream extends Stream
```

Comment:
It should not extend Stream at all. It is not compatible with any of its methods.

Reply:
I agree here, extending `Stream` makes no sense as we have to handle almost every method in a specific way.
Especially due to the `GdImage`/`resource` problematic, we could also just provide `GdImageStream` as it is for `GdImage` only. 
Not sure if some1 wants to use this for anything else, so renaming it might be helpful but I do not insist on this.